torchao/prototype/float8nocompile/float8nocompile_linear.py

```diff
@@ -151,25 +152,31 @@ def _forward_no_ac(
         orig_input_shape = input_hp.shape
         input_hp = input_hp.reshape(-1, input_hp.shape[-1])
 
-        # output = input @ weight_t
+        # compute input_fp8_row_major for forward.
+        # precompute input_fp8_col_major for backward.
         input_fp8_row_major, input_fp8_col_major = ToFP8RowAndColumnMajor.apply(
             input_hp,
             config.cast_config_input.target_dtype,
             linear_mm_config,
             GemmInputRole.INPUT,
             kernel_algo,
         )
-        weight_t_fp8_col_major = ToFP8ColumnMajorT.apply(
+
+        # compute weight_fp8_t_col_major for forward.
+        # precompute weight_fp8_col_major for backward.
+        weight_fp8_col_major, weight_t_fp8_col_major = ToFP8ColumnMajorTAndNonT.apply(
```

Comment:
I don't think we actually want this, see https://github.com/pytorch/ao/pull/936 for context.

You can benchmark the production path with the `force_recompute_fp8_weight_in_bwd` setting turned on to look at the behavior we want when in the FSDP + up to 128 GPUs + no other parallelisms regime.

Reply:
I see, sounds good